.cloudbees/testing/action.yml

```diff
@@ -116,6 +131,27 @@ runs:
           | .deploy.helm.releases[0].skipBuildDependencies = env(BUILD_DEPENDENCIES) != "true"
           | .deploy.statusCheckDeadlineSeconds = env(TIMEOUT_SECONDS)
           ' /tmp/skaffold/skaffold.yaml
+
+        if [ -s "${resolved_registries}" ]; then
+          findChartInMirrors() {
+            cat ${resolved_registries} | while read chart_location
+            do
+              chart_location=oci://$(echo "$chart_location" | awk -F':' '{print $1}') # strip the tag
+              echo "Verifying chart location: $chart_location by running helm show"
+              if helm show chart "$chart_location" --version "$CHART_VERSION"; then
+                echo "Setting chart location to $chart_location"
+                export CHART_LOCATION="$chart_location"
+                exit 0
```

Comment:
That needs to be `return 0`. Calling `exit 0` here exits the whole process.

Reply:
`return 0` was only exiting the loop since the way the file was being processed with a `|`, it was spawning a sub-shell. I updated the file processing code to not do that.